path_trie.go

```diff
@@ -21,6 +21,14 @@ func NewPathTrie() *PathTrie {
 	}
 }
 
+// NewPathTrieWithSegmenter allocates and returns a new *PathTrie with the given StringSegmenter implementation
+// other than PathSegmenter.
+func NewPathTrieWithSegmenter(segmenter StringSegmenter) *PathTrie {
```

Comment:
I'm fine with `func NewPathTrie` accepting a `type PathTrieConfig struct` with a single `StringSegmenter` field. This library doesn't do releases, so those using it via modules will be picking a sha and those opting to use latest can update or pin. Passing nil should retain the same default though.

Reply:
sure, thanks :) I believe other folks may benefit from this extension mechanism too.

Reply:
looks like today's GitHub fails to pick up the follow up fix. Would you mind to review further on pull request #16?